pennylane/compiler/qjit_api.py

```diff
@@ -105,8 +107,177 @@ def circuit(theta):
     """
 
     if not available(compiler):
-        raise RuntimeError(f"The {compiler} package is not installed.")
+        raise CompileError(f"The {compiler} package is not installed.")
 
     compilers = AvailableCompilers.names_entrypoints
     qjit_loader = compilers[compiler]["qjit"].load()
     return qjit_loader(fn=fn, *args, **kwargs)
+
+
+def while_loop(*args, compiler="catalyst", **kwargs):
```

Comment:
Does it make sense that each operation has a configurable compiler?
I thought this would be determined by the `qjit` decorator.

Reply:
I almost left the same comment, until I realized -- even _outside_ of `@qjit`, `qml.while_loop()` should still dispatch to the Catalyst version, which in turn will fallback to a Python implementation.

Reply:
Maybe later we can move the Python implementation of `while_loop` etc. to this dispatcher function, as the fallback if `@qjit` is not active.

Reply:
yeah, I also was wondering the same thing while developing the code. I don't have any argue with determining the compiler in the `qjit` decorator but as Josh mentioned, I though that this is a more general way of implementing these methods. However, there's no support for these `compiler` in particular and we can remove them for this soft-release.